pkg/metrics/statsd/statsd_test.go

```diff
@@ -83,3 +83,220 @@ func TestEmitFloat(t *testing.T) {
 		t.Error(m)
 	}
 }
+
+func TestParseSocketEnv(t *testing.T) {
+	for _, tt := range []struct {
+		name        string
+		teststring  string
+		wantNetwork string
+		wantAddress string
+		wantError   string
+	}{
+		{
+			name:        "Valid string",
+			teststring:  "foo:bar",
+			wantNetwork: "foo",
+			wantAddress: "bar",
+		},
+		{
+			name:        "Empty network part-regarded valid",
+			teststring:  ":bar",
+			wantNetwork: "",
+			wantAddress: "bar",
+		},
+		{
+			name:        "Empty address part-regarded valid",
+			teststring:  "foo:",
+			wantNetwork: "foo",
+			wantAddress: "",
+		},
+		{
+			name:       "No separator",
+			teststring: "somerandomtext",
+			wantError:  "malformed definition for the mdm statds socket. Expecting udp:<hostname>:<port> or unix:<path-to-socket> format. Got: \"somerandomtext\"",
+		},
+		{
+			name:       "Empty string",
+			teststring: "",
+			wantError:  "malformed definition for the mdm statds socket. Expecting udp:<hostname>:<port> or unix:<path-to-socket> format. Got: \"\"",
+		},
+		{
+			name:        "More than one separator",
+			teststring:  "a:b:c",
+			wantNetwork: "a",
+			wantAddress: "b:c",
+		},
+		{
+			name:        "Many separators",
+			teststring:  ":::::",
+			wantNetwork: "",
+			wantAddress: "::::",
+		},
+		{
+			name:        "One separator only",
+			teststring:  ":",
+			wantNetwork: "",
+			wantAddress: "",
+		},
+		{
+			name:        "Convert Upper Case network to lower case but leave address alone",
+			teststring:  "FOO:BAR",
+			wantNetwork: "foo",
+			wantAddress: "BAR",
+		},
```

Comment:
For visibility: it is a bit weird that we do not error on these cases, but we discussed it with @UlrichSchlueter and decided - it is not worth putting a lot of effort into validating something which is provided by us (SREs configuring the env).